cb-sol-cli/cmd/bridge.js

```diff
@@ -113,6 +113,112 @@ const queryResourceId = new Command("query-resource")
+    .option('--handler <address>', 'Handler address', constants.ERC20_HANDLER_ADDRESS)
+    .option('--targetContract <address>', `Contract address to be registered`, constants.ERC20_ADDRESS)
+    .option('--resourceId <address>', `Resource ID to be registered`, constants.ERC20_RESOURCEID)
+    .action(async function (args) {
+        await setupParentArgs(args, args.parent.parent)
+
+        const bridgeInstance = new ethers.Contract(args.bridge, constants.ContractABIsOptimism.Bridge.abi, args.wallet);
+        log(args,`Registering contract ${args.targetContract} with resource ID ${args.resourceId} on handler ${args.handler}`);
+        const tx = await bridgeInstance.adminSetResource(args.handler, args.resourceId, args.targetContract, { gasPrice: constants.OPTIMISM_GAS_PRICE, gasLimit: constants.OPTIMISM_GAS_LIMIT});
+        await waitForTx(args.provider, tx.hash)
+    })
+
+const registerGenericResourceCmdOptimism = new Command("register-generic-resource-optimism")
+    .description("Register a resource ID with a generic handler")
+    .option('--bridge <address>', 'Bridge contract address', constants.BRIDGE_ADDRESS)
+    .option('--handler <address>', 'Handler contract address', constants.GENERIC_HANDLER_ADDRESS)
+    .option('--targetContract <address>', `Contract address to be registered`, constants.CENTRIFUGE_ASSET_STORE_ADDRESS)
+    .option('--resourceId <address>', `ResourceID to be registered`, constants.GENERIC_RESOURCEID)
+    .option('--deposit <string>', "Deposit function signature", EMPTY_SIG)
+    .option('--execute <string>', "Execute proposal function signature", EMPTY_SIG)
+    .option('--hash', "Treat signature inputs as function prototype strings, hash and take the first 4 bytes", false)
+    .action(async function(args) {
+        await setupParentArgs(args, args.parent.parent)
+
+        const bridgeInstance = new ethers.Contract(args.bridge, constants.ContractABIsOptimism.Bridge.abi, args.wallet);
+
+        if (args.hash) {
+            args.deposit = getFunctionBytes(args.deposit)
+            args.execute = getFunctionBytes(args.execute)
+        }
+
+        log(args,`Registering generic resource ID ${args.resourceId} with contract ${args.targetContract} on handler ${args.handler}`)
+        const tx = await bridgeInstance.adminSetGenericResource(args.handler, args.resourceId, args.targetContract, args.deposit, args.execute, { gasPrice: constants.OPTIMISM_GAS_PRICE, gasLimit: constants.OPTIMISM_GAS_LIMIT})
+        await waitForTx(args.provider, tx.hash)
+    })
+
+const setBurnCmdOptimism = new Command("set-burn-optimism")
+    .description("Set a token contract as burnable in a handler")
+    .option('--bridge <address>', 'Bridge contract address', constants.BRIDGE_ADDRESS)
+    .option('--handler <address>', 'ERC20 handler contract address', constants.ERC20_HANDLER_ADDRESS)
+    .option('--tokenContract <address>', `Token contract to be registered`, constants.ERC20_ADDRESS)
```

Comment:
The description for `--tokenContract` should probably be consistent across all commands